src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -1163,20 +1209,31 @@ private static ExecutorService getExecutorService() {
         }
     }
 
-    private PeriodicShardSyncStrategy createPeriodicShardSyncStrategy(IKinesisProxy kinesisProxy,
-            ILeaseManager<KinesisClientLease> leaseManager) {
-        return new PeriodicShardSyncStrategy(
-                new PeriodicShardSyncManager(config.getWorkerIdentifier(), leaderDecider,
-                        new ShardSyncTask(kinesisProxy, leaseManager, config.getInitialPositionInStreamExtended(),
-                                config.shouldCleanupLeasesUponShardCompletion(),
-                                config.shouldIgnoreUnexpectedChildShards(), SHARD_SYNC_SLEEP_FOR_PERIODIC_SHARD_SYNC,
-                                shardSyncer, null), metricsFactory));
+    private PeriodicShardSyncStrategy createPeriodicShardSyncStrategy() {
+        return new PeriodicShardSyncStrategy(createPeriodicShardSyncManager());
     }
 
     private ShardEndShardSyncStrategy createShardEndShardSyncStrategy(ShardSyncTaskManager shardSyncTaskManager) {
         return new ShardEndShardSyncStrategy(shardSyncTaskManager);
     }
 
+    private PeriodicShardSyncManager createPeriodicShardSyncManager() {
+        if (leaderDecider == null) {
```

Comment:
I don't think we should ever allow `null` as a value for this to begin with, is there a case where this is valid?

Reply:
The entire KCL 1.x uses nulls instead of Optionals. Not ideal, but I think it's better to stick with the existing convention for the purposes of this PR